homeassistant/components/gtfs/sensor.py

```diff
@@ -308,123 +382,303 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     add_entities([
         GTFSDepartureSensor(gtfs, name, origin, destination, offset,
                             include_tomorrow)])
+    return True
 
 
 class GTFSDepartureSensor(Entity):
-    """Implementation of an GTFS departures sensor."""
+    """Implementation of a GTFS departure sensor."""
 
-    def __init__(self, pygtfs, name, origin, destination, offset,
-                 include_tomorrow) -> None:
+    def __init__(self, pygtfs: Any, name: Optional[Any], origin: Any,
+                 destination: Any, offset: cv.time_period,
+                 include_tomorrow: cv.boolean) -> None:
         """Initialize the sensor."""
         self._pygtfs = pygtfs
         self.origin = origin
         self.destination = destination
         self._include_tomorrow = include_tomorrow
         self._offset = offset
         self._custom_name = name
+
+        self._available = False
         self._icon = ICON
         self._name = ''
         self._state = None
-        self._attributes = {}
+        self._attributes = {}  # type: dict
+
+        self._agency = None
+        self._departure = {}  # type: dict
+        self._destination = None
+        self._origin = None
+        self._route = None
+        self._trip = None
+
         self.lock = threading.Lock()
         self.update()
 
     @property
-    def name(self):
+    def name(self) -> str:
         """Return the name of the sensor."""
         return self._name
 
     @property
-    def state(self):
+    def state(self) -> str:
         """Return the state of the sensor."""
+        if self._state is None:
+            return STATE_UNKNOWN
```

Comment:
Return `None` to represent unknown state.

Reply:
Returning `None` however introduces an inconsistency regarding Entity's state property return type:
https://github.com/home-assistant/home-assistant/blob/e70931da67381664a1bfb1bae10a914f7b9780b6/homeassistant/helpers/entity.py#L109-L113

Sort of modifying Entity's state property to return **Optional[str]**, I could instead define GTFS's state property as Optional[str], but also invalidates the interfacing somewhat. Hence my returning of **STATE_UNKNOWN**.

I'm curious, which cleanest path should we take?

Reply:
We should `return None` and let the base entity class handle the unknown state conversion.